src/pyaro/timeseries/Filter.py

```diff
@@ -855,4 +858,83 @@ def filter_stations(self, stations: dict[str, Station]) -> dict[str, Station]:
         if self._max_altitude is not None:
             stations = {n: s for n, s in stations.items() if (not math.isnan(s["altitude"]) and s["altitude"] <= self._max_altitude) }
         
-        return stations
\ No newline at end of file
+        return stations
+    
+@registered_filter
+class RelativeAltitudeFilter(StationFilter):
+    """
+    Filter class which filters stations based on the relative difference between
+    the station altitude, and the model topography altitude.
+
+    https://github.com/metno/pyaro/issues/39
+    """
+    def __init__(self, topo_file: str | None = None, topo_var: str = "topography", rtol: float = 1):
+        """
+        :param topo_file : A .nc file from which to read model topography data.
+        :param topo_var : Name of variable that stores altitude.
+        :param rtol : Relative toleranse.
+
+        Note:
+        -----
+        Stations will be kept if abs(altobs-altmod) <= rtol*abs(altobs)
+        """
+        self._topo_file = topo_file
+        self._topo_var = topo_var
+        self._rtol = rtol
+
+        self._topography = None
+        if topo_file is not None:
+            self._topography = xr.open_dataset(topo_file)
+        else:
+            logger.warning("No topography data provided (topo_file='%s'). Relative elevation filtering will not be applied.", topo_file)
+
+    def _model_altitude_from_lat_lon(self, lat: float, lon: float) -> float:
+        # TODO: Include a tolerance?
+        data = self._topography.sel({"lat": lat, "lon": lon}, method="nearest")
```

Comment:
`"lat"` and `"lon"` hardcoded? Can longitude/latitude names should be derived from the variables->coordinates (axes with units comparable to deg_north/deg_east according to CF-convention).